src/App.tsx

```diff
@@ -1,19 +1,124 @@
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos, addTodo, markDone, removeTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+// import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isHidden, setIsHidden] = useState(true);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [notCompleted, setNotCompleted] = useState(0);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const onChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = (
+    event: React.FormEvent,
+  ) => {
+    event.preventDefault();
+
+    setTodoTitle('');
+
+    const addNewTodo = async (data: Omit<Todo, 'id'>) => {
+      const todo = await addTodo(data);
+
+      setTodos((currentTodos): any => [...currentTodos, todo]);
+    };
+
+    if (user === null) {
+      return;
+    }
+
+    addNewTodo({
+      userId: user.id,
+      title: todoTitle,
+      completed: false,
+    });
+  };
+
+  const handleClick = async (id: number, completed: boolean) => {
+    await markDone(id, { completed: !completed });
+    setTodos((currentTodos): Todo[] => [...currentTodos]);
+  };
+
+  const handleRemoveTodo = async (id: number) => {
+    await removeTodo(id);
+  };
+
+  const hideError = () => {
+    setIsHidden(true);
+  };
+
+  const getFilter = (value: string) => {
+    setFilterStatus(value);
+  };
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    const loadTodosUser = async () => {
+      if (user === null) {
+        return;
+      }
+
+      try {
+        const todosUser = await getTodos(user.id);
+
+        switch (filterStatus) {
+          case 'all':
+            setTodos(todosUser);
+            setNotCompleted((
+              todosUser.filter(
+                todo => todo.completed === false,
+              )).length);
+            break;
+
+          case 'active':
+            setTodos(todosUser.filter(todo => todo.completed === false));
+            break;
+
+          case 'completed':
+            setTodos(todosUser.filter(todo => todo.completed === true));
+            break;
+
+          default:
+            break;
+        }
+      } catch (error) {
+        setIsHidden(false);
+      }
+    };
+
+    loadTodosUser();
+  }, [filterStatus]);
+
+  useEffect(() => {
+    const hide = setTimeout(() => {
+      setIsHidden(true);
+    }, 3000);
+
+    return () => clearTimeout(hide);
```

Comment:
```suggestion
    const hide = setTimeout(() => {
      setIsHidden(true);
    }, 3000);

    return hide;
```

then you can use the returned value whenever you want to clear the timeout (pressing the x button)
don't see what for you use this now